src/controller/python/chip/clusters/command.cpp

```diff
@@ -114,8 +145,32 @@ class CommandSenderCallback : public CommandSender::Callback
         delete this;
     };
 
+    CHIP_ERROR GetIndexFromCommandRef(uint16_t aCommandRef, size_t & aIndex)
+    {
+        auto search = mCommandRefToIndex.find(aCommandRef);
+        if (search == mCommandRefToIndex.end())
+        {
+            return CHIP_ERROR_KEY_NOT_FOUND;
+        }
+        aIndex = mCommandRefToIndex[aCommandRef];
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR AddCommandRefToIndexLookup(uint16_t aCommandRef, size_t aIndex)
+    {
+        auto search = mCommandRefToIndex.find(aCommandRef);
+        if (search != mCommandRefToIndex.end())
+        {
+            return CHIP_ERROR_DUPLICATE_KEY_ID;
+        }
+        mCommandRefToIndex[aCommandRef] = aIndex;
+        return CHIP_NO_ERROR;
+    }
+
 private:
     PyObject mAppContext = nullptr;
+    std::unordered_map<uint16_t, size_t> mCommandRefToIndex;
```

Comment:
This is guaranteed to never end up in an embedded device, right?

Reply:
This file is the interface for python-repl only as far as I am aware.